### DISCUSSION TITLE
Searching for more efficient block definition

### DISCUSSION THREAD

**Discussion by patins1** [deepjavalibrary/djl#1119] [Q&A]
With DJL, I want to write a ‘LambdaBlock’ that takes an NDList as input and returns a same-shaped NDList setting all inputs to 1 (assuming the NDList has only one entry).

The best I came up with is:

new LambdaBlock(x-> new NDList(x.singletonOrThrow().sub(x.singletonOrThrow()).add(1f)))

Is there a way to achieve the same without using subtraction operation (so it would be executed faster)?

**Comment by zachgk** [comment]
Use `x.onesLike()` ​([javadoc](https://javadoc.io/doc/ai.djl/api/latest/ai/djl/ndarray/NDArray.html#onesLike--))

Also, you can use `LambaBlock.singleton(...)` to make it a bit cleaner ([javadoc](https://javadoc.io/static/ai.djl/api/0.12.0/ai/djl/nn/LambdaBlock.html#singleton-java.util.function.Function-))
